lib/resources/collection/index.js

```diff
@@ -637,11 +649,18 @@ function createDomain(data, errors) {
   return domain;
 }
 
-Collection.prototype.doAfterCommitEvent = function(method, ctx, data){
+Collection.prototype.doAfterCommitEvent = function(method, ctx, data, previous, protectedKeys) {
   var collection = this;
   if (collection.shouldRunEvent(collection.events.AfterCommit, ctx)) {
-    collection.events.AfterCommit.run(ctx, {data: data, 'this': data, method: method}, function (err) { 
-      if (err) debug('AfterCommit errors in script: %j', err); 
+    data = _.clone(data);
+    if (protectedKeys && protectedKeys.length > 0 && previous) {
+      // add back whatever fields were protected, because they are removed from data
+      protectedKeys.forEach(function (key) {
+        data[key] = previous[key];
+      });
+    }
```

Comment:
Not sure why that would be preferrable. The code is related to the AfterCommit event and I think it makes more sense to be inside that function rather than adding more code to the already long enough caller.